tests/test_statespace/test_sde.py

```diff
@@ -232,6 +244,28 @@ def test_solve_mde_forward_values(ltisde_as_linearsde, ltisde, v_const, diffusio
     np.testing.assert_allclose(out_linear.cov, out_lti.cov)
 
 
+def test_solve_mde_forward_sqrt_values(
+    ltisde_as_linearsde,
+    ltisde_as_linearsde_sqrt_forward_implementation,
+    v_const,
+    diffusion,
+):
+    """mde forward values in sqrt-implementation and classic implementation should be equal"""
+    out_linear, _ = ltisde_as_linearsde.forward_realization(
+        v_const, t=0.0, dt=0.1, _diffusion=diffusion
+    )
+
+    out_linear_2, _ = ltisde_as_linearsde.forward_rv(
+        out_linear, t=0.1, dt=0.1, _diffusion=diffusion
+    )
+    out_linear_2_sqrt, _ = ltisde_as_linearsde_sqrt_forward_implementation.forward_rv(
+        out_linear, t=0.1, dt=0.1, _diffusion=diffusion
+    )
+
+    np.testing.assert_allclose(out_linear_2_sqrt.mean, out_linear_2.mean)
+    np.testing.assert_allclose(out_linear_2_sqrt.cov, out_linear_2.cov)
```

Comment:
Please correct me if I am wrong. The sqrt-forward implementation solves a different ODE than the non-square root implementation, both use an adaptive solver. So is it actually meaningful to check that the outputs are identical up to numerical precision? It is already tested whether they coincide with the LTI transition, right? 

I am not sure whether I am asking a question or proposing anything. But do you think that it would be a meaningful layer of testing to do the present test here on a non-lti sde?

Reply:
This should test the forward values being the same for both implementations. I thought for the simple LTI case they should coincide. Nevertheless, I think testing on a non-LTI SDE might be a good idea.

Reply:
> It is already tested whether they coincide with the LTI transition, right?
If I understand you correctly, it is already tested whether the classic implementation coincides with the LTI transition but not whether the square-root implementation coincides with the LTI transition.

Reply:
Ah I see. In this case nevermind my comment :) Thanks!